src/V3Options.cpp

```diff
@@ -135,6 +135,75 @@ VTimescale::VTimescale(const string& value, bool& badr)
     }
 }
 
+//######################################################################
+// V3HierarchyBlockOption class functions
+
+// Parse "--hierarchy-block orig_name,mangled_name,param0_name,param0_value,... " option.
+// The format of value is as same as -G option. (can be string literal surrounded by ")
+V3HierarchyBlockOption::V3HierarchyBlockOption(const string& opts) {
+    V3StringList vals;
+    bool inStr = false;
+    string cur;
+    // Split by ','. If ',' appears between "", that is not a separator.
+    for (string::const_iterator it = opts.begin(); it != opts.end();) {
+        if (inStr) {
+            if (*it == '\\') {
+                ++it;
+                UASSERT(it != opts.end(), " must not end with \\");
+                UASSERT(*it == '"' || *it == '\\', *it << " can not appear");
```

Comment:
Still think these need to be v3error's instead of asserts.

Reply:
3eda905cac31f21b7b60354ee54976f3f0e654db uses v3error.